features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/ViewOnDrawInterceptor.kt

```diff
@@ -44,16 +45,33 @@ internal class ViewOnDrawInterceptor(
     }
 
     fun stopIntercepting() {
-        decorOnDrawListeners.entries.forEach {
-            it.key.viewTreeObserver.removeOnDrawListener(it.value)
+        decorOnDrawListeners.entries.forEach { (decorView, listener) ->
+            if (decorView.viewTreeObserver.isAlive) {
```

Comment:
seems we have this check already in the `stopInterceptingSafe` anyway, it can be removed here

Reply:
Good catch